cmd/notation/plugin.go

```diff
@@ -62,3 +108,169 @@ func listPlugins(command *cobra.Command) error {
+	if err != nil {
+		return err
+	}
+
+	pluginName := newPlugin["name"].(string)
+	newPluginVersion := newPlugin["version"].(string)
+	newSemVersion, err := semver.NewVersion(newPluginVersion)
+	if err != nil {
+		return err
+	}
+
+	// get plugin directory
+	pluginDir, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		panic(err)
+	}
+
+	// Check if plugin directory exists
+	_, err = os.Stat(pluginDir)
+
+	// create the directory, if not exist
+	if os.IsNotExist(err) {
+		if err := os.MkdirAll(pluginDir, 0755); err != nil {
+			return err
+		}
+	}
+
+	// Check if plugin exists
+	_, err = os.Stat(pluginDir + "/" + pluginBinary)
+
+	// copy plugin, if not exist
+	if os.IsNotExist(err) {
+		copyPlugin(pluginPath, pluginDir+"/"+pluginBinary)
+	}
+
+	// overwrite plugin, if force flag is set
+	if err == nil && force {
+		fmt.Printf("Overwriting plugin %s in directory %s\n", pluginBinary, pluginDir)
+		copyPlugin(pluginPath, pluginDir+"/"+pluginBinary)
+	}
+
+	// if plugin exists and force flag is not set, get plugin metadata
+	if err == nil && !force {
+		cmd := exec.Command(pluginDir+"/"+pluginBinary, "get-plugin-metadata")
+
+		output, err := cmd.Output()
+		if err != nil {
+			return err
+		}
+
+		var currentPlugin map[string]interface{}
+		err = json.Unmarshal([]byte(output), &currentPlugin)
+		if err != nil {
+			return err
+		}
+
+		// check if new plugin version is greater than current plugin version
+		currentPluginVersion := currentPlugin["version"].(string)
+		currentVersion, err := semver.NewVersion(currentPluginVersion)
+		if err != nil {
+			return err
+		}
+
+		// copy plugin, if new plugin version is greater than current plugin version
+		if newSemVersion.GreaterThan(currentVersion) {
+			var confirm string
+
+			fmt.Printf("Detected new version %s. Current version is %s.\nDo you want to overwrite the plugin %s? [y/N]: ", newSemVersion.String(), currentVersion.String(), pluginName)
+			fmt.Scanln(&confirm)
+
+			if strings.ToLower(confirm) != "y" {
+				fmt.Println("Operation cancelled.")
+				return nil
+			}
+
+			fmt.Printf("Copying plugin %s to directory %s...\n", pluginName, pluginDir)
+			copyPlugin(pluginPath, pluginDir+"/"+pluginBinary)
+		}
+
+		// do not copy plugin, if new plugin version is less than or equal to current plugin version
+		if newSemVersion.LessThan(currentVersion) || newSemVersion.Equal(currentVersion) {
+			fmt.Println("Current version is greater than or equal to new version. Skipping plugin installation.\nUse --force flag to overwrite the plugin.")
+		}
+	}
+
+	return nil
+}
+
+func removePlugin(command *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin name")
+	}
+
+	pluginName := args[0]
+
+	// get plugin directory
+	pluginDir, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		return err
+	}
+
+	// Check if plugin directory exists
+	_, err = os.Stat(pluginDir)
+	if os.IsNotExist(err) {
+		return errors.New("plugin does not exist")
+	}
```

Comment:
@priteshbandi good point. What do you think about a helper called `checkPluginDir` that handles this logic and is used in the install and remove funcs?

Reply:
I added a helper called exists to reduce the duplicate code. It might be a helpful function to add into the internal ioultil package at some point. Same with the `splitPluginName` that I added.